pkg/apis/core/types.go

```diff
@@ -3510,17 +3510,42 @@ type LoadBalancerIngress struct {
 	Hostname string
 }
 
+const (
+	// MaxServiceTopologyKeys is the largest number of topology keys allowed on a service
+	MaxServiceTopologyKeys = 16
+)
+
 // IPFamily represents the IP Family (IPv4 or IPv6). This type is used
-// to express the family of an IP expressed by a type (i.e. service.Spec.IPFamily)
+// to express the family of an IP expressed by a type (e.g. service.spec.ipFamilies).
 type IPFamily string
 
 const (
 	// IPv4Protocol indicates that this IP is IPv4 protocol
 	IPv4Protocol IPFamily = "IPv4"
 	// IPv6Protocol indicates that this IP is IPv6 protocol
 	IPv6Protocol IPFamily = "IPv6"
-	// MaxServiceTopologyKeys is the largest number of topology keys allowed on a service
-	MaxServiceTopologyKeys = 16
+)
+
+// IPFamilyPolicyType represents the dual-stack-ness requested or required by a Service
+type IPFamilyPolicyType string
+
+const (
+	// SingleStack indicates that this service is required to have a single IPFamily.
+	// The IPFamily assigned is based on the default IPFamily used by the cluster
+	// or as identified by service.spec.ipFamilies field
+	SingleStack IPFamilyPolicyType = "SingleStack"
```

Comment:
The vast majority of constants in `corev1` have some sort of "type" prefix in the constant name

	ProtocolTCP Protocol = "TCP"

	TerminationMessageFallbackToLogsOnError TerminationMessagePolicy = "FallbackToLogsOnError"

	ConditionTrue ConditionStatus = "True"

So this should probably be

	IPFamilySingleStack IPFamilyPolicyType = "SingleStack"

Reply:
This is *not* ignored. It would be easier to do that once we are done with everything else.